.github/workflows/build.yml

```diff
@@ -0,0 +1,119 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron: '0 4 * * *' # Runs every day at 4am: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#scheduled-events-schedule
+
+jobs:
+
+  build:
+    name: ${{ matrix.os }}, Node.js v${{ matrix.node }}, Python ${{ matrix.python }}
+
+    strategy:
+      matrix:
+        os: [windows-2019, ubuntu-18.04, macos-10.15]
+        node: ['12.x']
+        python: ['2.x']
+        include:
+          - os: ubuntu-18.04
+            node: '12.x'
+            python: '3.x'
+            tests: 'skip'
+
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # To fetch all history for all branches and tags. (Will be required for caching with lerna: https://github.com/markuplint/markuplint/pull/111)
+
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Use Python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Build
+        shell: bash
+        run: |
+          yarn --skip-integrity-check --network-timeout 100000
+          npx electron-replace-ffmpeg
+          npx electron-codecs-test
+          ./scripts/check_git_status.sh
+        env:
+          NODE_OPTIONS: --max_old_space_size=4096
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # https://github.com/microsoft/vscode-ripgrep/issues/9
+
+      - name: Test (headless)
+        if: matrix.tests != 'skip'
+        shell: bash
+        run: |
+          yarn test:theia
+
+      - name: Test (browser)
+        if: matrix.tests != 'skip' && runner.os == 'Linux'
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: yarn test:browser
+
+      - name: Test (electron)
+        if: matrix.tests != 'skip' && runner.os == 'Linux'
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: yarn test:electron
+
+  publish:
+    needs: build
+    if: github.ref == 'refs/heads/master' && github.event_name != 'schedule' # We still publish the manually dispatched workflows: 'workflow_dispatch'.
```

Comment:
So no more `next` publish on each commit from `master`, but every day at 4am?

Reply:
I wanted to publish when on `master` and **not** a CRON job (e.g., merged PRs.). Where is the bug? I must be blind 😊

Reply:
Ok I misread the conditional. Can you explain why we want to run the tests at 4am? I did not see a note about this in the PR description.

Reply:
I understand that this `publish` job runs when we push on master and not when the cron runs, but why the cron at all?

Reply:
> but why the cron at all?

I do not know if we have one currently, but we used to have a CRON job from the beginning of this project. I set it up on Travis UI if I remember correctly. [Here](https://travis-ci.org/github/eclipse-theia/theia/builds/691440689) is one build example from 6 months ago.

Reply:
> I do not know if we have one currently, but we used to have a CRON job from the beginning of this project.

Ok but why? I can't find a reason for it or anything that requires a daily job. If we publish on each commit on master as well as test, then what would change from day to day that needs the CI to run for?